--- apt-1.4.8/apt-pkg/deb/dpkgpm.cc	2017-09-13 11:47:33.000000000 -0500
+++ apt-sileo/apt-pkg/deb/dpkgpm.cc	2018-12-25 23:31:59.377765073 -0600
@@ -1697,7 +1697,7 @@
    bool dpkgMultiArch = debSystem::SupportsMultiArch();
 
    // start pty magic before the loop
-   StartPtyMagic();
+   //StartPtyMagic(); or not...
 
    // Tell the progress that its starting and fork dpkg
    d->progress->Start(d->master);
@@ -2096,7 +2096,7 @@
       }
    }
    // dpkg is done at this point
-   StopPtyMagic();
+   //StopPtyMagic();
    CloseLog();
 
    if (d->dpkg_error.empty() == false)
