--- apt-1.4.8/CMake/Misc.cmake	2017-09-13 11:47:33.000000000 -0500
+++ apt-sileo/CMake/Misc.cmake	2018-12-25 23:31:59.357764959 -0600
@@ -66,16 +66,16 @@
 
 # Generates a simple version script versioning everything with current SOVERSION
 function(add_version_script target)
-    get_target_property(soversion ${target} SOVERSION)
-    set(script "${CMAKE_CURRENT_BINARY_DIR}/${target}.versionscript")
-    string(REPLACE "-" "" name "${target}_${soversion}")
-    string(TOUPPER "${name}" name)
-    add_custom_command(OUTPUT "${script}"
-                       COMMAND echo "${name} {global: *; };" > "${script}"
-                       VERBATIM )
-    add_custom_target(${target}-versionscript DEPENDS "${script}")
-    target_link_libraries(${target} PRIVATE -Wl,-version-script="${script}")
-    add_dependencies(${target} ${target}-versionscript)
+	#get_target_property(soversion ${target} SOVERSION)
+    	#set(script "${CMAKE_CURRENT_BINARY_DIR}/${target}.versionscript")
+    	#string(REPLACE "-" "" name "${target}_${soversion}")
+    	#string(TOUPPER "${name}" name)
+    	#add_custom_command(OUTPUT "${script}"
+    	#                   COMMAND echo "${name} {global: *; };" > "${script}"
+    	#                   VERBATIM )
+    	#add_custom_target(${target}-versionscript DEPENDS "${script}")
+    	#target_link_libraries(${target} PRIVATE -Wl,-version-script="${script}")
+    	#add_dependencies(${target} ${target}-versionscript)
 endfunction()
 
 function(path_join out path1 path2)
