--- apt-1.7.0/apt-pkg/contrib/progress.cc	2018-10-07 00:56:02.000000000 -0500
+++ apt-sileo/apt-pkg/contrib/progress.cc	2018-12-25 23:53:03.055945371 -0600
@@ -119,7 +119,7 @@
    auto const Now = std::chrono::steady_clock::now().time_since_epoch();
    auto const Now_sec = std::chrono::duration_cast<std::chrono::seconds>(Now);
    auto const Now_usec = std::chrono::duration_cast<std::chrono::microseconds>(Now - Now_sec);
-   struct timeval NowTime = { Now_sec.count(), Now_usec.count() };
+   struct timeval NowTime = { Now_sec.count(), static_cast<__darwin_suseconds_t>(Now_usec.count()) };
 
    std::chrono::duration<decltype(Interval)> Delta =
       std::chrono::seconds(NowTime.tv_sec - LastTime.tv_sec) +
