--- apt-1.4.8/test/interactive-helper/test_fileutl.cc	2017-09-13 11:47:33.000000000 -0500
+++ apt-sileo/test/interactive-helper/test_fileutl.cc	2018-12-25 23:31:59.521765891 -0600
@@ -14,7 +14,7 @@
 
 static void callsystem(std::string const &call)
 {
-   auto ret = system(call.c_str());
+   auto ret = RunCmd(call.c_str());
    if (WIFEXITED(ret) == false || WEXITSTATUS(ret) != 0)
       _error->Error("Calling %s failed!", call.c_str());
 }
